internal/impl/laravel/5.8/handler/generator/transactor.go

```diff
@@ -0,0 +1,173 @@
+package generator
+
+import (
+	"asher/internal/api"
+	"asher/internal/api/codebuilder/php/builder"
+	"asher/internal/api/codebuilder/php/builder/interfaces"
+	"asher/internal/api/codebuilder/php/core"
+	"github.com/iancoleman/strcase"
+	"strings"
+)
+
+type TransactorGenerator struct {
+	classBuilder   interfaces.Class
+	identifier     string
+	imports        []string
+	transactorType string
+}
+
+func NewTransactorGenerator(identifier string, transactorType string) *TransactorGenerator {
+	return &TransactorGenerator{
+		classBuilder:   builder.NewClassBuilder(),
+		identifier:     identifier,
+		imports:        []string{},
+		transactorType: transactorType,
+	}
+}
+
+/**
+Sets the identifier of the current class
+Parameters:
+	- identifier: string
+Sample Usage:
+	- SetIdentifier("ClassName")
+*/
+func (transactorGenerator *TransactorGenerator) SetIdentifier(identifier string) *TransactorGenerator {
+	transactorGenerator.identifier = identifier
+	return transactorGenerator
+}
+
+/**
+Sets the type of the transactor
+Parameters:
+	- identifier: string
+Sample Usage:
+	- SetTransactorType("default") or SetTransactorType("file") or SetTransactorType("image")
+*/
+func (transactorGenerator *TransactorGenerator) SetTransactorType(identifier string) *TransactorGenerator {
+	transactorGenerator.transactorType = identifier
+	return transactorGenerator
+}
+
+/**
+Appends import to the controller file
+Parameters:
+	- units: string array of the import
+Returns:
+	- instance of ControllerGenerator object
+Sample Usage:
+	- AppendImport([]string{"App\User",})
+*/
+func (transactorGenerator *TransactorGenerator) AppendImports(units []string) *TransactorGenerator {
+	transactorGenerator.imports = append(transactorGenerator.imports, units...)
+	return transactorGenerator
+}
+
+/**
+Adds Constructor in the Transactor with Query and Mutator Injected of the currentModel
+Returns:
+	- Return instance of TransactorGenerator
+Sample Usage:
+	- transactorGeneratorObject.AddConstructorFunction()
+*/
+func (transactorGenerator *TransactorGenerator) AddConstructorFunction() *TransactorGenerator {
+	lowerCamelIdentifier := strcase.ToLowerCamel(transactorGenerator.identifier)
+	queryVariableName := lowerCamelIdentifier + `Query`
+	mutatorVariableName := lowerCamelIdentifier + `Mutator`
+
+	constructorArguments := []string{
+		transactorGenerator.identifier + `Query $` + queryVariableName,
+		transactorGenerator.identifier + `Mutator $` + mutatorVariableName,
+	}
+
+	parentConstructorCall := core.NewFunctionCall("parent::__construct").
+		AddArg(core.NewParameter("$" + queryVariableName)).
+		AddArg(core.NewParameter("$" + mutatorVariableName)).
+		AddArg(core.NewParameter(`"id"`))
+
+	switch transactorGenerator.transactorType {
```

Comment:
move to handler... either make specific classes are just simply move this code over there